jujugui/static/gui/src/app/app.js

```diff
@@ -1767,6 +1767,10 @@ YUI.add('juju-gui', function(Y) {
         }
         charmstore.getEntity(url.legacyPath(), callback);
       };
+      const getModelName = () => {
+        const name = this.db.environment.get('name');
+        return name === 'untitled-model' ? undefined : name;
```

Comment:
I think you could just use `this.env.get('environmentName')` as it doesn't get set to "untitled-model".